django/core/signing.py

```diff
@@ -150,11 +155,23 @@ class Signer(object):
 
     def __init__(self, key=None, sep=':', salt=None):
         # Use of native strings in all versions of Python
-        self.sep = force_str(sep)
+        self.sep = sep
         self.key = key or settings.SECRET_KEY
         self.salt = force_str(salt or
             '%s.%s' % (self.__class__.__module__, self.__class__.__name__))
 
+    @property
+    def sep(self):
+        return self._sep
+
+    @sep.setter
+    def sep(self, value):
+        sep = force_str(value)
+        if _SEP_UNSAFE.match(sep):
+            warnings.warn('Unsafe Signer separator: %r (cannot be empty or consist of only A-z0-9-_=)' % sep,
```

Comment:
sep is used in as an argument to `rsplit` in `unsign` like so:

    value, sig = signed_value.rsplit(self.sep, 1)

So as long as the separator contains at least one character that will never appear in the signature string it should be safe.

I can remove the `setter`. My reasoning behind it is that `Signer().sep` is (undocumented, but) public, so it should be checked at all times. Same could be said for key and salt. I do agree that it bloats the code a fair bit. So I guess it's your call.